stl/inc/format

```diff
@@ -2557,8 +2557,221 @@ template <_Format_supported_charT _CharT, class _Traits>
 struct formatter<basic_string_view<_CharT, _Traits>, _CharT>
     : _Formatter_base<basic_string_view<_CharT, _Traits>, _CharT, _Basic_format_arg_type::_String_type> {};
 
-using format_context  = basic_format_context<back_insert_iterator<string>, string::value_type>;
-using wformat_context = basic_format_context<back_insert_iterator<wstring>, wstring::value_type>;
+template <class _T>
+class _Fmt_buffer {
+private:
+    _T* _Ptr;
+    size_t _Size;
+    size_t _Capacity;
+
+protected:
+    _Fmt_buffer(size_t _Sz) noexcept : _Size(_Sz), _Capacity(_Sz) {}
+    _Fmt_buffer(_T* _P = nullptr, size_t _Sz = 0, size_t _Cap = 0) noexcept : _Ptr(_P), _Size(_Sz), _Capacity(_Cap) {}

```

Comment:
I think with the default member initializer we can drop the default arguments

Reply:
Also avoiding macro like names. Recently the form `_Ptr_` as the argument for the member `_Ptr` has seen a lot of use
```suggestion
    _Fmt_buffer(_Ty* _Ptr_, const size_t _Size_, const size_t _Capacity_) noexcept : _Ptr(_Ptr_), _Size(_Size_), _Capacity(_Capacity_) {}
```

Reply:
All of these should be changed now.